src/flag_gems/ops/conv1d.py

```diff
@@ -0,0 +1,140 @@
+            + (tl.arange(0, BLOCK_C_IN) * padded_width_input)[:, None]
+            + i * stride_width
+            + tl.arange(0, BLOCK_W)
+        )
+        mask_input = (offset_n)[:, None, None] < n_out and (tl.arange(0, BLOCK_C_IN))[
+            :, None
+        ] < c_input
+        weight_offset = (
+            (tl.arange(0, BLOCK_C_IN) * width_kernel * c_out)[:, None, None]
+            + (tl.arange(0, BLOCK_W) * c_out)[:, None]
+            + offset_cout
+        )
+        mask_weight = (offset_cout) < c_out and (tl.arange(0, BLOCK_C_IN))[
+            :, None, None
+        ] < c_input
+
+        input_value = tl.load(padded_input + input_offset, mask=mask_input, other=0)
+        weight_value = tl.load(weight + weight_offset, mask=mask_weight, other=0)
+        input_value = tl.reshape(input_value, (BLOCK_N, BLOCK_OUT_WEIGHT))
+        weight_value = tl.reshape(weight_value, (BLOCK_OUT_WEIGHT, BLOCK_O))
+        accumulator = tl.dot(input_value, weight_value, allow_tf32=False)
+        out_offset = (
+            (offset_n * c_out * width_out)[:, None] + (offset_cout) * width_out + i
+        )
+        mask_out = (offset_n)[:, None] < n_out and offset_cout < c_out
+        tl.store(out + out_offset, accumulator, mask=mask_out)
+
+
+class Conv1d(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, input, weight, bias, stride, padding, dilation, groups):
+        logging.debug("GEMS CONV1D")
+        assert input.ndim == 2 or input.ndim == 3, "Invalid input tensor."
+        assert weight.ndim == 2 or weight.ndim == 3, "Invalid weight tensor."
+        if input.ndim == 3:
+            n_out, c_input, width_input = list(input.shape)
+        else:
+            c_input, width_input = list(input.shape)
+            n_out = 1
+
+        if weight.ndim == 3:
+            c_out, c_kernel, width_kernel = list(weight.shape)
+            weight = weight.permute(1, 2, 0).contiguous()
+        else:
+            c_kernel, width_kernel = list(weight.shape)
+            c_out = 1
+        assert c_kernel == c_input, "Invalid channel value in input and weight"
+        # todo make sure tuple is 2d
+        if isinstance(stride, list):
+            stride_width = stride[0]
+        else:
+            stride_width = stride
+
+        if isinstance(padding, list):
+            padding_width = padding[0]
+        else:
+            padding_width = padding
+
+        width_out = (width_input + 2 * padding_width - width_kernel) // stride_width + 1
+
+        padded_input = torch.nn.functional.pad(
+            input, (padding_width, padding_width), "constant", 0
+        )
+        padded_input = padded_input.contiguous()
+        _, _, padded_width_input = list(padded_input.shape)
+        out = torch.zeros(
+            n_out, c_out, width_out, dtype=input.dtype, device=input.device
+        )
+        grid = lambda meta: (
+            triton.cdiv(n_out, meta["BLOCK_N"]),
+            triton.cdiv(c_out, meta["BLOCK_O"]),
```

Comment:
suggest using (n_out, c_out) as grid, since 3d pointer in triton has problems unsolved.